nft-barter-core/test/nft-barter.js

```diff
@@ -343,4 +335,25 @@ contract("NFT Barter", (accounts) => {
       ERROR_PERMISSION_DENIED
     );
   });
+
+  it("isSwapPossible", async () => {
+    const isSwapPossible = await tokenInstance.isSwapPossible.call(swapId);
+    assert.isTrue(isSwapPossible);
+
+    await tokenInstance.acceptSwap(swapId, takerTokenId, {
+      from: takerAddress,
+    });
+
+    const swapShouldNotBePossible = await tokenInstance.isSwapPossible.call(
+      swapId
+    );
+    assert.isFalse(swapShouldNotBePossible);
+  });
+
+  it("isSwapPossible - failure cases", async () => {
```

Comment:
don't understand what's happening here.

Reply:
We should also test for behavior where tokens in a swap are transferred to some other address. 
1) initiate a swap with token-1 and token-2 where addr1 owns token-1 and addr2 owns token-2
2) token-1 is transferred to add-3 OR token-2 is transferred to addr-3
3) Swap should not be possible It should be a useless swap.

Reply:
Nice catch. Added test for it.

Reply:
It still doesn't cover the scenario I have described above. 
Please take note that in the 2nd step you have to transfer through the simple `tranferFrom` method of ERC-721.